experimental/dds/tree2/src/test/feature-libraries/contextuallyTyped.spec.ts

```diff
@@ -81,10 +97,115 @@ describe("ContextuallyTyped", () => {
 		const numberSequence = SchemaBuilder.fieldSequence(numberSchema);
 		const primaryObject = builder.object("numbers", { local: { [EmptyKey]: numberSequence } });
 		const schema = builder.intoDocumentSchema(numberSequence);
-		const mapTree = applyTypesFromContext(schema, new Set([primaryObject.name]), []);
+		const mapTree = applyTypesFromContext(
+			{ schema, getFieldGenerator: defaultGetFieldGenerator },
+			new Set([primaryObject.name]),
+			[],
+		);
 		const expected: MapTree = { fields: new Map(), type: primaryObject.name, value: undefined };
 		assert.deepEqual(mapTree, expected);
 	});
 
+	describe("cursorFromContextualData adds globalFieldKey", () => {
+		it("for empty contextual data.", () => {
+			const builder = new SchemaBuilder("Identifier Domain");
+			const identifierSchema = builder.primitive("identifier", ValueSchema.String);
+			const identifierFieldSchema = builder.globalField(
+				"identifier",
+				SchemaBuilder.fieldValue(identifierSchema),
+			);
+			const nodeSchema = builder.objectRecursive("node", {
+				local: {
+					child: SchemaBuilder.fieldRecursive(FieldKinds.optional, () => nodeSchema),
+				},
+				global: [identifierFieldSchema] as const,
+			});
+
+			const nodeSchemaData = builder.intoDocumentSchema(
+				SchemaBuilder.fieldOptional(nodeSchema),
+			);
+			const globalFieldKey: GlobalFieldKey = brand("identifier");
+			const contextualData: ContextuallyTypedNodeDataObject = {};
+
+			const getFieldGenerator = (
+				key: FieldKey,
+				schema: FieldStoredSchema,
+			): FieldGenerator => {
+				const fieldGenerator = (): MapTree[] => {
+					return [
+						{
+							value: uuid(),
```

Comment:
for this test, its probably better to use a simple hard coded string like "x" rather than generating a UUID. Then you can easily assert that the generated field is correct, not just that its not undefined.

Reply:
updated `fieldGenerator` to return hardcoded value 'x'